azure-pipelines.yml

```diff
@@ -196,16 +201,38 @@ jobs:
     inputs:
       targetType: inline
       script: |
+        if (-NOT ([Security.Principal.WindowsPrincipal][Security.Principal.WindowsIdentity]::GetCurrent()).IsInRole([Security.Principal.WindowsBuiltInRole] "Administrator")) {  
+          # Start another Powershell process as Admin and execute this script again
+          $arguments = "& '" +$myinvocation.mycommand.definition + "'"
+          Start-Process powershell -Verb runAs -ArgumentList $arguments
+          # Stop if the PowerShell is not run as Admin
+          Break
+        }
+        # The following are executed by elevated PowerShell
+        az --version
+
         $InstallArgs = @(
-          "/I"
-          '"$env:SYSTEM_ARTIFACTSDIRECTORY/msi/Microsoft Azure CLI.msi"'
+          "/i"
+          "`"$env:SYSTEM_ARTIFACTSDIRECTORY\msi\Microsoft Azure CLI.msi`""
+          "/q"
           "/norestart"
-          "/L*v"
+          "/l*v"
           ".\install_logs.txt"
         )
+        $pre_installed_version=az version --query '\"azure-cli\"' -o tsv
+        $to_be_installed_version=Get-Content $(System.ArtifactsDirectory)/metadata/version
+        if ($pre_installed_version -eq $to_be_installed_version){
+          # See https://docs.microsoft.com/windows/win32/msi/reinstallmode about options of REINSTALLMODE
+          $reinstall_option="REINSTALL=ALL REINSTALLMODE=emus" 
+          $InstallArgs += $reinstall_option
+        }
```

Comment:
ADO agent has a pre-installed Azure CLI MSI, so this job tries to "reinstall" with the artifact MSI. If the artifact MSI has the same version as the pre-installed Azure CLI, L226 adds `REINSTALL=ALL REINSTALLMODE=emus`.

If I run the `msiexec.exe` command locally (with or without `/q`):

```cmd
msiexec.exe /i "Microsoft Azure CLI.msi" REINSTALL=ALL REINSTALLMODE=emus
```

it succeeds very fast but doesn't install anything, so the "reinstall" actually never works.

I guess it is because the artifact MSI is actually a different product as the pre-installed one, so it has nothing to "reinstall". I am not sure if this is related to `Product.Id` being `*`:

https://github.com/Azure/azure-cli/blob/aeb94ed12a46acdd14e7c4e2633908e149f4a1bc/build_scripts/windows/Product.wxs#L36

After removing `REINSTALL=ALL REINSTALLMODE=emus`, the artifact MSI can install successfully, but is installed side-by-side with the pre-installed one.